### ISSUE TITLE
Hover tooltip not showing for step line

### ISSUE THREAD

**Issue by mjmare** [bokeh/bokeh#7419]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
Python 3.6
Bokeh 0.12.13

#### Description of expected behavior and the observed behavior
Using the default hover tool I do not see the tooltip when using the step line. Switching to circle or line will result to the correct behaviour.

#### Complete, minimal, self-contained example code that reproduces the issue

```
from bokeh.plotting import figure, output_file, show

# output to static HTML file
output_file("test.html")

p = figure(plot_width=400,
           plot_height=400,
           tools='hover')

# p.circle([1, 2, 3, 4, 5], [6, 7, 2, 4, 5],
#          size=20,
#          color="navy",
#          alpha=0.5,
#          )

p.step([1, 2, 3, 4, 5], [6, 7, 2, 4, 5],
         color="navy",
         alpha=0.5,
         )

# p.line([1, 2, 3, 4, 5], [6, 7, 2, 4, 5],
#        color="navy",
#        alpha=0.5,
#        )

# show the results
show(p)
```

**Comment by bryevdv** [comment]
@cloud-rocket that error was fixed in a recent PR, tho note that I specifically mean the error was spurious error was suppressed. Note that adding hit testing support to step (so that hover works on it) is still an open issue (it just won't error out anymore)

**Comment by sdaza** [comment]
Any news on this?

**Comment by saurabhkumar** [comment]
Python 3.6.8
Bokeh 1.2.0

I am not able to use the hover tool with the step function (there is no error but the tooltip does not show up). Is there any alternative possible if the problem is not fixed in the current version? My problem is that I have a timeseries data (discrete values) and if I use the line function, the hover tool gives interpolated values in between two points which are not meaningful.

**Comment by bryevdv** [comment]
> I am not able to use the hover tool with the step function (there is no error but the tooltip does not show up). 

Right, it has not been implemented yet. The previous PR was just to keep things from blowing up if a step and hover tool were present at the same time. 

>  Is there any alternative possible if the problem is not fixed in the current version

There might be some ways, but none of them are great. E.g. you could potentially use `segment` to draw the horizontal parts of the step line. These would just be to drive hit testing, so they could be made invisible (i.e. set `alpha=0`).

**Comment by jeisch** [comment]
In version 1.4.0, the list under "Hover tool does not currently work with the following glyphs:" does not include any reference to step.

**Comment by bryevdv** [comment]
@jeisch A PR go update the docs would be an excellent way for a new contributor to make a substantive change that takes burden off the (few) core contributors. Please let us know if we a can help you make a PR.

**Comment by Cyberfly100** [comment]
I would like to express interest in this feature as well. We are trying to visualize histograms with many bins (>10k) with bokeh. bars/boxes seem to perform way worse and are not as feature-rich in general than the step glyph, it seems. We prefer the step glyph over the line, since, when zoomed in close, it is still clear what the size of a single bin is. However, due to the hover feature, we are using line now.

**Comment by a-gertner** [comment]
I am in no position to demand features from core development team, I know resources are limited. But can someone please share the status of this? Is this possible / not possible / planned / not a priority / ... ?

Thanks

**Comment by bryevdv** [comment]
@a-gertner Unfortunately, the step glyph is different from almost every other glyph in that it has to internally synthesize fairly different coordinates for drawing than it starts with, which presents both technical and conceptual challenges fitting in with existing assumptions about hit-testing and selections. Sometimes features languish, and this seems to be one of those instances. I can't give you any estimate on when this might be implemented.

**Comment by a-gertner** [comment]
@bryevdv , Thanks a lot for finding the time to reply. I totally understand the constraints.
I looked at the code briefly but unfortunately I don't feel competent enough to submit a PR. I will just leave it for now.

**Comment by debegr92** [comment]
As a workaround you could add a dot plot to the chart. Working good for me.

**Comment by a-gertner** [comment]
> As a workaround you could add a dot plot to the chart. Working good for me.

@debegr92, can you explain a bit more or provide an example? Thanks

**Comment by debegr92** [comment]
> provide an example

Sure! Just add a dot plot to your step plot. Hover tool will work on the dots and this workaround is fine for me.

```python
import pandas as pd
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
from bokeh.models import ColumnDataSource
output_notebook()

x = [1,2,3,4,5]
y = [0,1,0,1,0]

TOOLTIPS = [
  ("x", "@x"),
  ("y", "@y")
]

cds = ColumnDataSource(pd.DataFrame({"x":x, "y":y}))

p1 = figure(height=300, sizing_mode="stretch_width", title="Step Hover Test", tooltips=TOOLTIPS)
p1.step(x="x", y="y", mode="after", source=cds)

# Helper to show hover with step plot
p1.dot(x="x", y="y", source=cds)

show(p1)
```

![Image](https://github.com/user-attachments/assets/1b9102b1-3804-418f-a32d-afa568ba1913)

**Comment by a-gertner** [comment]
> Sure! Just add a dot plot to your step plot. Hover tool will work on the dots and this workaround is fine for me.

While this is clearly a workaround and it has its limitations, I agree that this is a great idea. I have tweaked it a bit:

```python
import pandas as pd
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
from bokeh.models import ColumnDataSource, HoverTool
output_notebook()

x = [1,2,3,4,5,6,7,8,9]
y = [0,1,0,1,0,1,1,0,0]

TOOLTIPS = [
  ("x", "@x"),
  ("y", "@y")
]

cds = ColumnDataSource(pd.DataFrame({"x":x, "y":y}))

p1 = figure(height=300, sizing_mode="stretch_width", title="Step Hover Test")
p1.step(x="x", y="y", mode="after", source=cds)

# Helper to show hover with step plot
dots = p1.dot(x="x", y="y", source=cds)

hover_tool = HoverTool(tooltips=TOOLTIPS, mode='vline')
p1.add_tools(hover_tool)

show(p1)
```

This increases the hit area of the HoverTool to not just the dot but anywhere on the vertical line passing thru the dot
